t/integration/test_canvas.py

```diff
@@ -232,6 +233,30 @@ def test_ready_with_exception(self):
         while not result.ready():
             pass
 
+
+    @flaky
+    def test_not_ready_with_http_exception(self):
+        g = group([add.s(1, 2), raise_custom_error.s()])
+        result = g.apply_async()
+        with pytest.raises(TypeError) as e:
```

Comment:
Just so I'll get this right, this **should** fail with a TypeError right?
Instead, it raises a HTTPError?

Reply:
No, the opposite.
The expected behaviour, should be that raise_custom_error tasks since it raises explicitly the HTTPError to be propagated as it is to the join method.
Instead a TypeError is raised stating that the HTTPError is not called with all 5 mandatory arguments.
And the raise_custom_error as you can see it raises the HTTPError correctly.